gwctl/pkg/printer/backends.go

```diff
@@ -28,7 +36,101 @@ import (
 )
 
 type BackendsPrinter struct {
-	Out io.Writer
+	Out   io.Writer
+	Clock clock.Clock
+}
+
+func (bp *BackendsPrinter) Print(resourceModel *resourcediscovery.ResourceModel) {
+	tw := tabwriter.NewWriter(bp.Out, 0, 0, 2, ' ', 0)
+	row := []string{"NAMESPACE", "NAME", "TYPE", "REFERRED BY ROUTES", "AGE", "POLICIES"}
+	_, err := tw.Write([]byte(strings.Join(row, "\t") + "\n"))
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to write to the tab writer: %v\n", err)
+		os.Exit(1)
+	}
+
+	backendNodes := make([]*resourcediscovery.BackendNode, 0, len(resourceModel.Backends))
+	for _, backendNode := range resourceModel.Backends {
+		backendNodes = append(backendNodes, backendNode)
+	}
+
+	sort.Slice(backendNodes, func(i, j int) bool {
+		if backendNodes[i].Backend.GetNamespace() != backendNodes[j].Backend.GetNamespace() {
+			return backendNodes[i].Backend.GetNamespace() < backendNodes[j].Backend.GetNamespace()
+		}
+		return backendNodes[i].Backend.GetName() < backendNodes[j].Backend.GetName()
+	})
+
+	for _, backendNode := range backendNodes {
+		backend := backendNode.Backend
+
+		parentHTTPRoutes := []string{}
+		remainderHTTPRoutes := 0
+		associatedDistinctPolicies := map[string]any{}
+
+		httpRouteNodes := make([]*resourcediscovery.HTTPRouteNode, len(backendNode.HTTPRoutes))
+		i := 0
+		for _, node := range backendNode.HTTPRoutes {
+			httpRouteNodes[i] = node
+			i++
+		}
+		sort.Slice(httpRouteNodes, func(i, j int) bool {
+			if httpRouteNodes[i].HTTPRoute.GetNamespace() != httpRouteNodes[j].HTTPRoute.GetNamespace() {
+				return httpRouteNodes[i].HTTPRoute.GetNamespace() < httpRouteNodes[j].HTTPRoute.GetNamespace()
+			}
+			return httpRouteNodes[i].HTTPRoute.GetName() < httpRouteNodes[j].HTTPRoute.GetName()
+		})
+
+		for _, httpRouteNode := range httpRouteNodes {
+			httpRoute := httpRouteNode.HTTPRoute
+
+			if len(parentHTTPRoutes) < 2 {
+				namespacedName := client.ObjectKeyFromObject(httpRoute).String()
+				parentHTTPRoutes = append(parentHTTPRoutes, namespacedName)
+			} else {
+				remainderHTTPRoutes++
+			}
+
+			// TODO(yashvardhan-kukreja): change this to httpRouteNode.EffectivePolicies once the work around inherited policies is matured
```

Comment:
You should be counting the policies which are directly attached to the Backend: `BackendNode.Policies` (might need to adjust the tests as well)

Reply:
@gauravkghildiyal I tried doing this but the output seemed a bit weird.

None of the backendRefs got any policies associated with them with the above method.

Can you give an example of a directly attached policy to a backend? i.e. which resource can I apply to simulate directly attaching policy to a backend.